### PR TITLE
[FLINK-13437][test] Add Hive SQL E2E test

### PR DESCRIPTION
**Pull request by zjuwangg**
## What is the purpose of the change
Set up a docker-based yarn-cluster and hive service using the new java based test runtime framework, add HiveConnectorITCase to cover data read/write function, including:
  1. hive data writen by Hive, read by Flink.
  2. hive data writen by Flink, read by Hive.
  3. read/write to a non-partition table.
  4. multi-format for read and write, cover textfile/orc/parquet
Based on this PR, we can add more test such as function/view in further more.

## Brief change log

  - 3488ec6 Add e2e test for hive data connector using docker based environment
  - 2f8b127 refactor hive e2e test using new java-based test framework
  - 76c6f08 add muliti format test and all data types test case
  - 31cc4b7 remote e2e bash test

## Verifying this change
  - *Added integration tests for end-to-end deployment *

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable )

### CODE REVIEW

**Comment by zjuwangg** [comment]
cc @bowenli86 @xuefuz @JingsongLi @KurtYoung @lirui-apache to have a review

**Comment by zjuwangg** [comment]
It's a base work, and we can add more ITCase based on this PR.

**Comment by zjuwangg** [comment]
@bowenli86 @JingsongLi Do you guys have time to have a basic look?

File: flink-end-to-end-tests/flink-connector-hive-test/src/test/java/org/apache/flink/connectors/hive/HiveConnectorITCase.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		try (final ClusterController clusterController = flinkResource.startCluster(1)) {
+			JobSubmission.JobSubmissionBuilder jobSubmissionBuilder = new JobSubmission.JobSubmissionBuilder(testJarPath);
+			jobSubmissionBuilder.setParallelism(1)
+					.addOption("-ys", "1")
+					.addOption("-ytm", "1000")
+					.addOption("-yjm", "1000")
+					.addOption("-c", HiveReadWriteDataExample.class.getCanonicalName())
+					.addArgument("--hiveVersion", hiveVersion)
+					.addArgument("--sourceTable", "all_types_table")
+					.addArgument("--targetTable", "dest_all_types_table");
```
Comment (lirui-apache) [inline flink-end-to-end-tests/flink-connector-hive-test/src/test/java/org/apache/flink/connectors/hive/HiveConnectorITCase.java]: I think the job submission code is the same for the 2 test cases. Can we reuse it?

**Comment by bowenli86** [comment]
@lirui-apache @JingsongLi can you guys help review? I can help merge once it passes

**Comment by JingsongLi** [comment]
Thanks @zjuwangg for your great work! I will continue to working on this.

**Comment by sjwiesman** [comment]
What's the status of this?

**Comment by JingsongLi** [comment]
> What's the status of this? With the new sql filesystem connector I suspect more flink users will rely on Hive integration. It would be good to try and get this in for 1.12.

Yes, we should move on.
I plan to modify testing to sql-client but don't have enough time in 1.11, we should finish this in 1.12.

**Comment by zjuwangg** [comment]
Close this comment!

**Pull request closed without merging**